sormas-app/app/src/main/java/de/symeda/sormas/app/LbdsRecevierComponent.java

```diff
@@ -0,0 +1,234 @@
+package de.symeda.sormas.app;
+
+import java.lang.reflect.InvocationTargetException;
+import java.nio.charset.StandardCharsets;
+import java.security.PublicKey;
+import java.util.List;
+
+import org.hzi.sormas.lbds.core.http.HttpContainer;
+import org.hzi.sormas.lbds.core.http.HttpMethod;
+import org.hzi.sormas.lbds.core.http.HttpResult;
+import org.hzi.sormas.lbds.messaging.Constants;
+import org.hzi.sormas.lbds.messaging.IntentType;
+import org.hzi.sormas.lbds.messaging.IntentTypeCarrying;
+import org.hzi.sormas.lbds.messaging.LbdsPropagateKexToSormasIntent;
+import org.hzi.sormas.lbds.messaging.LbdsResponseIntent;
+import org.hzi.sormas.lbds.messaging.util.KeySerializationUtil;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.googlecode.openbeans.IntrospectionException;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.util.Base64;
+import android.util.Log;
+
+import androidx.annotation.Nullable;
+
+import de.symeda.sormas.api.PushResult;
+import de.symeda.sormas.api.caze.CaseDataDto;
+import de.symeda.sormas.api.person.PersonDto;
+import de.symeda.sormas.app.backend.caze.Case;
+import de.symeda.sormas.app.backend.caze.CaseDao;
+import de.symeda.sormas.app.backend.common.DaoException;
+import de.symeda.sormas.app.backend.common.DatabaseHelper;
+import de.symeda.sormas.app.backend.config.ConfigProvider;
+import de.symeda.sormas.app.backend.person.Person;
+import de.symeda.sormas.app.backend.person.PersonDao;
+import de.symeda.sormas.app.lbds.LbdsDtoHelper;
+import de.symeda.sormas.app.rest.RetroProvider;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class LbdsRecevierComponent extends IntentService {
+
+	/**
+	 * Creates an IntentService. Invoked by your subclass's constructor.
+	 */
+	public LbdsRecevierComponent() {
+		super("LbdsRecevierComponent");
+		// super(SormasRelated.SORMAS_RECEIVER_COMPONENT);
+	}
+
+	@Override
+	protected void onHandleIntent(@Nullable Intent intent) {
+		Log.i("SORMAS_LBDS", "==========================");
+		final String intentType = intent.getStringExtra(Constants.INTENT_TYPE);
+		Log.i("SORMAS_LBDS", "Received LBDS intent: " + intentType);
+		if (intentType != null && !intentType.trim().isEmpty()) {
+			IntentType type = IntentType.valueOf(intentType);
+			switch (type) {
+			case KEX_TO_LBDS_INTENT:
+				final String publicKey = intent.getStringExtra(Constants.SORMAS_KEY);
+				final PublicKey key = KeySerializationUtil.deserializePublicKey(publicKey);
```

Comment:
I guess this is the reply coming from LBDS that the key was received. Anything to do with this?

Reply:
Can be removed

Reply:
done